### PR TITLE
0.1.15/refactor/new wireframes

### PR DESCRIPTION
**Pull request by ArnoldasSid**
Refactor of screens 2 and 3

### CODE REVIEW

**Comment by spastai** [comment]
Tests are failing. 
But it's not the worst thing in this PR. 

We should clearly agree on architecture (and Redux use).
Previously state was bound to URL, now we are moving it to redux. The worst thing is that in some cases state is in URL and for some dialogs - in redux.
Should we agree that Redux is only for such cases when one view is calling other view and comes back to initial view?

**Comment by ArnoldasSid** [comment]
About tests - only recurrent trip tests are failing, I'll add a commit with the tests removed as we have agreed that we will not have this feature (at least in the nearby future).

About application state:
tl;dr
I strongly disagree with keeping state in url in general (I see no real advantages to it)

To begin with redux is explicitly designed to hold application state and in my opinion does much better job than url's. Keeping the state in a single immutable place really helps out maintainability wise (also redux helps a lot with debuging things, as it's easy to log what has happened and reproduce things).
Also the way I thnk about our app is that it's first and foremost a mobile application which also happens to live on the web. And given that mobile applications are pretty fine without the urls, I don't think our application needs to be strongly tied to urls too.
The main reason to have urls in my opinion is so that user can share the link with a friend, however the only legitimate place to do so in our application is to send a link to some trip to your friend. That's why I like the idea of having url of "/trip/:id" instead of "/rideRequest/:id", "/driveConfirm/:id" and so on. But we already have that part broken, now you can't really share links to trip with your friend because your friend might have a different relationship to the trip than you do. It's architecture/maintainability vs user needs. Admittedly gains are pretty minimal either way.
Also another nice thing we gain by using redux is that application logic get's moved from views to actions. Now for example views don't know and don't need to know how trips are created (be it carpool service or in whataver other way) and what extra params carpool service needs from global application state to create the trip. To sum up I believe that we should keep everything, that is not local to some component, in redux stores unless there is a strong reason to do otherwise.

File: src/packages/carpool-view-flow/react/screens/rides-list-with-data/RidesListWithDataScreen.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { createContainer } from 'meteor/react-meteor-data';
+import RidesList from '../../components/rides-list/RidesList';
+
+/*global Progress*/
+/*global carpoolService*/
+
+export default createContainer(({filterOwn = "all", role = "driver", aLoc, bLoc, bTime}) => {
```
Comment (spastai) [inline src/packages/carpool-view-flow/react/screens/rides-list-with-data/RidesListWithDataScreen.jsx:7]: Why this appears in screens/ 
Screens suppose to be agnostic view rendering components.

**Comment by spastai** [comment]
1. Strong argument for URL is from maintenance perspective - adding tools like google analytics provide a lot of information with zero cost. Of course good logging with redux or analytics events use would produce similar result - but you have to implement those things.
2. Code should be uniform. And architecture: container (Meteor)/state manager+actions (Redux)/view (React) should be described or clearly visible from the code and used everythere (see comment for [RidesListWithDataScreen](https://github.com/vilnius/vilnius-carpool/pull/203/files#diff-d3e32aa50f05f0c91c8afe6816b492c1R1) 
3. I propose to keep both - urls (it should be improved) and redux states. 
   We should investigate bind url and actions in declarative way

**Comment by spastai** [comment]
http://localhost:3000/m/your/newRide
Gives:
modules.js?hash=be3f2bd…:21208 Warning: Failed propType: Invalid prop `bTime` of type `object` supplied to `TripForm`, expected `number`. Check the render method of `Connect(TripForm)`.warning 

modules.js?hash=be3f2bd…:21208 Warning: Failed propType: Invalid prop `progress` of type `object` supplied to `TripForm`, expected `number`. Check the render method of `Connect(TripForm)`.

modules.js?hash=be3f2bd…:21208 Warning: Failed propType: Required prop `stops` was not specified in `TripForm`. Check the render method of `Connect(TripForm)`.

File: src/packages/carpool-view-flow/react/screens/trip-form/TripFormScreen.jsx [hunk 2]
```diff
@@ -112,54 +72,49 @@ class TripForm extends React.Component {
...
+  openDateTimePicker () {
+    this.picker.openDateTimePicker(this.state.isDepartureDate, this.props.tripDateTime)
   }
 
-  toInputSelect (val) {
-    this.setState({
-      to: val,
-    })
-  }
+  // submitForm () {
```
Comment (spastai) [inline src/packages/carpool-view-flow/react/screens/trip-form/TripFormScreen.jsx]: Could you add line 
// TODO <some action that should be done with those comments>

File: src/packages/carpool-view-flow/react/screens/trip-form/TripFormScreen.jsx [hunk 3]
```diff
@@ -321,13 +271,23 @@ TripForm.propTypes = {
...
 
+const connectedTripForm = connect((state) => ({
+  locationFrom: locationFromSelector(state),
+  locationTo: locationToSelector(state),
+  tripDateTime: tripDateTimeSelector(state),
+}))(TripForm);
+
 export default createContainer(() => {
   const progress = new Progress();
   const stops = carpoolService.pullStops(progress.setProgress.bind(progress, 'stops'));
```
Comment (spastai) [inline src/packages/carpool-view-flow/react/screens/trip-form/TripFormScreen.jsx:288]: I should add some comments here - stops are/were needed to be loaded as path routing was done in client.
But I think it could be removed from TripForm.propTypes
In general those stops are required only when creatingDrive

**Comment by ArnoldasSid** [comment]
About RidesListWithDataScreen.jsx

I'm not sure what you mean - Is the problem because we don't have a screen component wrapping it? It would probably be good to do so for consistency. But I think it's ok as it is. I think "if it ain't broken don't fix it" mentality is good here because there is enough of things that are broken.

It is a copy of what currently is at https://github.com/vilnius/vilnius-
carpool/blob/develop/src/packages/carpool-view-flow/react/screens/ride-offers/RideOffersScreen.jsx

This file was written by you, and is used by multiple screens (not just ride offers) in router.js, however I needed RideOffers to look differently (new design). Therefore I changed RideOffersScreen.jsx and made new file for the rest of screens that use this.

About url's:
They definitely should be used where there is a reason to use them. However I'm strongly against them as a generic solution to hold application state, because, even though they get the job done for most static websites, it will get very hard to maintain in an interactive application.

About the later comments:
I guess I was too afraid of breaking things, so I didn't remove a bunch of code that likely should not be there anymore. The idea was to do a cleanup refactor later as everything gets refactored to redux now I'm only sure that TripForm and RideOffers screen work correctly. Might have been a mistake on my side... Should I clean it up for this PR?
The problem is that I'm heavily confused how this aLoc, bLoc stuff works with global variables and url parameters that are written to / read from god knows where, that's why I REALLY like redux :) )

**Comment by spastai** [comment]
1. RideOffersScreen
   Initially our screens like RideOffersScreen were smart and did business logic, actions handling and data loading. 
   Now we are splitting those functions into separate components, thus data loading components like would be reasonable to move RidesListWithDataScreen to the other folder.
   Do you agree that our client components diagram looks like:
   5cc-modules-redux
2. URLs
   As I wrote don't mind to put everything what is in variables now into redux. 
3. Big changes 
   Easier to handle in different branches. I would suggest to push this request into vilnius:0.1.15/refactor/new-wireframes branch instead of vilnius:develop
   Then deleting whole bunch of code is safer as you can quickly switch to the other branch.
   In addition I would be able to add changes on that branch as every PR should contain refactoring and it seems that our code is becoming more messy, so before merging into develop we should clean it.
